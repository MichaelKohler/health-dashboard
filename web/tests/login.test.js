/* global it, expect */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import { useSelector } from 'react-redux';
import { shallowToJson  } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Login from '../src/login.jsx';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));

it('renders correctly', () => {
  useSelector.mockImplementation(
    (selectorFn) => selectorFn({
      loginFailed: false,
    }),
  );

  const wrapper = shallow(<Login />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('renders correctly - login failed', () => {
  useSelector.mockImplementation(
    (selectorFn) => selectorFn({
      loginFailed: true,
    }),
  );

  const wrapper = shallow(<Login />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

/* eslint-enable react/jsx-filename-extension */
