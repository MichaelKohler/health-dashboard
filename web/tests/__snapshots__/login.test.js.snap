// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - login failed 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <CssBaseline />
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <div
    className="makeStyles-paper-3"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-7"
    >
      <LockOutlinedIcon />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
    <p>
      Login failed
    </p>
    <form
      className="makeStyles-form-5"
      noValidate={true}
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="email"
        fullWidth={true}
        id="email"
        label="Email Address"
        margin="normal"
        name="email"
        required={true}
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        autoComplete="current-password"
        fullWidth={true}
        id="password"
        label="Password"
        margin="normal"
        name="password"
        required={true}
        type="password"
        variant="outlined"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-6"
        color="primary"
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Log in
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <CssBaseline />
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <div
    className="makeStyles-paper-3"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-7"
    >
      <LockOutlinedIcon />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-5"
      noValidate={true}
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="email"
        fullWidth={true}
        id="email"
        label="Email Address"
        margin="normal"
        name="email"
        required={true}
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        autoComplete="current-password"
        fullWidth={true}
        id="password"
        label="Password"
        margin="normal"
        name="password"
        required={true}
        type="password"
        variant="outlined"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-6"
        color="primary"
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Log in
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;
